### ISSUE TITLE
ObjectHidden - Self referencing entity...

### ISSUE THREAD

**Issue by ThaDafinser** [doctrine/DoctrineModule#278]
I just want to send a hidden element to save a reference over a form.
The selection is done hidden, so the user won't select it anywhere. 

The problem is that i can't use the standard Zend\Form\Element\Hidden, because a proxy is generated. 

I don't want to use the ObjectSelect, because those queries are not necessary.

``` PHP
<?php
namespace LispLogbook\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use LispBase\Entity\AbstractEntity;
use LispApplication\Entity\CodePosition;

/**
 * @ORM\Table(name="pstEntry")
 * @ORM\Entity(repositoryClass="LispLogbook\Entity\EntryRepository")
 * @ORM\HasLifecycleCallbacks
 */
class Entry extends AbstractEntity
{

    /**
     * @ORM\Id
     * @ORM\Column(name="entId", type="guid")
     * @ORM\GeneratedValue(strategy="UUID")
     *
     * @var string
     */
    protected $id;


    /**
     * @ORM\ManyToOne(targetEntity="LispLogbook\Entity\Entry")
     * @ORM\JoinColumn(name="entParentId", referencedColumnName="entId")
     *
     * @var \LispLogbook\Entity\Entry
     */
    private $parent;

```

My current solution feels like a "hack", but works for me.

``` PHP
<?php
namespace DoctrineModule\Form\Element;

use DoctrineModule\Form\Element\Proxy;
use Zend\Form\Element\Hidden as HiddenElement;
use Zend\Form\Form;

class ObjectHidden extends HiddenElement
{
    /**
     * @var Proxy
     */
    protected $proxy;

    /**
     * @return Proxy
     */
    public function getProxy()
    {
        if (null === $this->proxy) {
            $this->proxy = new Proxy();
        }
        return $this->proxy;
    }

    /**
     * @param  array|\Traversable $options
     * @return ObjectSelect
     */
    public function setOptions($options)
    {
        $this->getProxy()->setOptions($options);
        return parent::setOptions($options);
    }

    /**
     * {@inheritDoc}
     */
    public function setValue($value)
    {
        return parent::setValue($this->getProxy()->getValue($value));
    }

    /**
     * {@inheritDoc}
     */
    public function getValueOptions()
    {
        if (empty($this->valueOptions)) {
            $this->setValueOptions($this->getProxy()->getValueOptions());
        }
        return $this->valueOptions;
    }
}
```

The call in the form fieldset:

``` PHP
        $this->add(array(
            'type' => 'DoctrineModule\Form\Element\ObjectHidden',
            'name' => 'parent',
            'options' => array(
                'object_manager' => $entityManager,
                'target_class' => 'LispLogbook\Entity\Entry'
            )
        ));
```

**Comment by mikemix** [comment]
Works good for me as well. I needed to include related object's ID in the hidden field since the form row is fetched using AJAX.

**Comment by TomHAnderson** [comment]
Unless anyone has more to day on this I'm going to file it as will not address because it causes a BC break.

**Comment by j-schumann** [comment]
I have nothing new on this but why will it not be addressed in 2.0.0? This new major version seems to still be in pre-release state and already contains other BC breaks?
